Remove the To-do URL

whit537 opened the pull request:
Closes #4166; part of #4130.

## Todo

- [x] use a branch.sql—https://github.com/gratipay/gratipay.com/pull/4214#discussion_r90640285
- [x] fix formatting niggle—https://github.com/gratipay/gratipay.com/pull/4214#discussion_r90640398

whit537 commented:
Okay @JessaWitzel, this one's for you ([ref](http://gratipay.slackarchive.io/gratipay/-/1480368214.000245/1480515513.000351/1480508012000345/))! :-)

Your [first PR](https://github.com/gratipay/inside.gratipay.com/pull/874) was for inside.gratipay.com ... this one is gonna be your first PR for gratipay.com itself! 💃

Here's what we are looking for here:

- [ ] spin up your local environment to the point where you can see and submit the "[Apply for a New Team](https://gratipay.com/new)" form locally—that's what you'll be working on here
- [ ] find and remove the "To-do URL" field from the form
- [ ] also remove the `todo_url` field from the `teams` table in the database
- [ ] make sure the tests still pass
- [ ] make sure the form still works

Does that give you enough to get started? For broader context, check the project linked in the sidebar here on GitHub (and check the "Details" on the project page). As always, if you're stuck after an hour, ask us a question here or on slack! Okay! Good luck! 👍 💃 🌻

JessaWitzel commented:
Have removed the to-do URL field from the form and the field from the teams table (I think) but when I try to create a team locally I receive gratipay.com/error.spt "Please fill out the 'To-do URL' field." Went into testing and removed the todo_url test but am still receiving error.

whit537 commented:
[slack](http://gratipay.slackarchive.io/gratipay/-/1480533275.000404/1480535223.000504/1480534290000464/)

mattbk commented:
This is how I like to solve problems like this: https://github.com/gratipay/gratipay.com/search?utf8=%E2%9C%93&q=Please+fill+out+the+%27To-do+URL%27+field.

JessaWitzel commented:
The tests worked for a minute but then I had to clean and re: make schema make fake and the fake data was wrong. Now I am receiving a subprocess error when I try to run tests that stalls on the first test related to honcho?

JessaWitzel commented:
Why does make test have so many more failures than make pytest? This passes when I make pytest.

JessaWitzel commented:
I ended up removing all references to todo_url in the code, not just those related to the form because it didn't pass tests until I did. I hope that is what you wanted. :)

whit537 commented:
!m @JessaWitzel 

I see the tasks are ticked in https://github.com/gratipay/gratipay.com/pull/4214#issuecomment-263888476. Does that mean you're done here? If so, go ahead and move it to the "Landing" column on the project (see sidebar) and then ping a few of us here by name to let us know it's ready for review! :)

JessaWitzel commented:
Ready for review friends! @whit537 @mattbk @clone1018

mattbk commented:
I got this when I submitted:

![screen shot 2016-12-01 at 7 50 16 pm](https://cloud.githubusercontent.com/assets/3729038/20820247/6f90c284-b7ff-11e6-8482-298a04ef173c.png)

JessaWitzel commented:
You know what? I fixed that and then after all of the testing errors I had to redo some things and never checked to see if I could submit the form. I will fix.

JessaWitzel commented:
After cleaning my environment and removing cached files, the form works for me. Can you try that? 
![image](https://cloud.githubusercontent.com/assets/21209540/20822015/27489d74-b815-11e6-9e97-03f29a634110.png)

mattbk commented:
Success after `make clean env`!

whit537 commented on tests/py/test_teams.py:
```diff
@@ -226,13 +225,10 @@ def test_casing_of_urls_survives(self):
-                          , onboarding_url='http://INSIDE.GRATipay.com/'
-                          , todo_url='hTTPS://github.com/GRATIPAY'
-                           ))
+                          , onboarding_url='http://INSIDE.GRATipay.com/'))
```
Sorry to be picky but can we match the parentheses formatting here as it was before? :)

whit537 replied:
Fixed in 6062804f4ce3e1a9719d82aff6ed4d4ee1b51078. :-)

whit537 commented:
Looks good so far, @JessaWitzel! Good work! 💃 

I was able to submit the form locally. 👍

I've left a couple todos (in the ticket description), one tiny and one more involved.

JessaWitzel commented on sql/schema.sql:
```diff
@@ -546,7 +546,6 @@ BEGIN;
     ALTER TABLE teams ALTER COLUMN getting_paid DROP NOT NULL;
 
     ALTER TABLE teams ADD COLUMN onboarding_url text NOT NULL DEFAULT '';
-    ALTER TABLE teams ADD COLUMN todo_url text NOT NULL DEFAULT '';
```
Apologies. There is so much to remember!

whit537 replied:
No apologies necessary! You're doing great! :-)

whit537 commented on sql/schema.sql:
```diff
@@ -889,6 +890,12 @@ BEGIN;
+--https://github.com/gratipay/gratipay.com/pull/4214
+BEGIN;
+  ALTER TABLE teams DROP COLUMN todo_url;
+END;
```
@JessaWitzel Close! Rather than modifying `sql/schema.sql` directly, what we want is a file at `sql/branch.sql` (it shouldn't exist on your branch; you have to create it and add it to Git). If there's a file at `sql/branch.sql` then our deploy script [takes care of](https://github.com/gratipay/gratipay.com/blob/2fcc8a36d89aae6ac46f91dd7fa8bfcc038b4d7a/deploy.sh#L66-L85) appending it to `sql/schema.sql` after running `sql/branch.sql` against the production database. Make sense?

whit537 commented:
Rebased; was 2fcc8a36d89aae6ac46f91dd7fa8bfcc038b4d7a.

whit537 commented:
@JessaWitzel Somehow we ended up with a bunch of commits from master on this branch. Probably something about the state of your local git working copy vis-a-vis the remote copy here on GitHub? Maybe something related to me switching the base branch from `master` to `relax-open-work-requirement`? One way or another, I've now rebased this branch on the latest `relax-open-work-requirement`. Just the four relevant commits are on here. You'll have to do a dance to get your local copy synced up with this one. Oftentimes I'll just delete my local branch and re-check-out from `origin` in order to get the latest; you can compare SHAs w/ `git log` locally and the web UI here on GitHub to make sure you're synced.

![screen shot 2016-12-05 at 11 52 34 am](https://cloud.githubusercontent.com/assets/134455/20893877/5502c3ba-bae1-11e6-960d-f469a8995623.png)

whit537 commented:
Here's an easy way to see the latest SHA locally:

```
$ git log | head -n1
commit c8c7914bec1be89e0fa6ca4079614be2e633a190
```

JessaWitzel commented:
@whit537 @mattbk @clone1018 hopefully ready to review finally! One day I will do these on the second try. :)

whit537 commented:
!m @JessaWitzel 

Now all we need is to back out the changes to schema.sql! :-)

whit537 commented:
You can either add a new commit that undoes the changes ... or if you wanna level up your git-fu, you could do an interactive rebase to remove the c8c7914bec1be89e0fa6ca4079614be2e633a190 commit. Make sure you're on ec2179673f9ef0ab20a7f906bf4b1431644c125a locally, and then:

```
git rebase -i head^^
```

That should pop up an editor with additional instructions.

It's totally fine to just make a new commit, too. :-)

whit537 commented:
Rebased onto the latest #4221; previous head was ec2179673f9ef0ab20a7f906bf4b1431644c125a. I've updated my previous comment to reflect the latest commits.

whit537 commented:
> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated

whit537 commented:
Aaaaaaaand #4224. 😩

whit537 commented:
Returning from #4224 ...

whit537 commented:
Builds restarted. Will it be as simple as that?

whit537 commented:
Green! Let's see if we can land this ...

whit537 commented:
Rebased, was 2596672a354b0fb4518971cda8024e816979d7a8.

whit537 commented:
Hit #4224. Cache cleared for this PR and build restarted ...

whit537 commented:
I removed the caches for `remove-todo-url` and `relax-open-work-requirement` for good measure. I left `master`.

whit537 commented:
Looks like it's working! 💃

whit537 commented:
I accidentally rebased 2596672a354b0fb4518971cda8024e816979d7a8 away! Brought it back with:

```
wget https://github.com/gratipay/gratipay.com/commit/2596672a354b0fb4518971cda8024e816979d7a8.patch
git am 2596672a354b0fb4518971cda8024e816979d7a8.patch
```

whit537 commented:
Woo-hoo! Good work, @JessaWitzel! 💃 🌻
